### PR TITLE
Include extension version in the library built for the extension

### PR DESCRIPTION
**Pull request by jonsnyder**
## Description

In the library built for the Launch extension, this changes the implementationDetails.name field to be "alloy+launch", and the implementationDetails.version field to be `${alloyVersion}+${extensionVersion}`. In Alloy we were using a rollup plugin to do the version replacement, but in the launch extension we are not using rollup. So to use the same code for both alloy and the launch extension, I found "babel-plugin-version" to do the version replacement. Alloy looks for \_\_VERSION\_\_ and the launch extension looks for \_\_EXTENSION_VERSION\_\_.

## Related Issue

https://jira.corp.adobe.com/browse/CORE-48374

## Motivation and Context

We want to know the version of the launch extension used to collect experience events.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

**Review by Aaronius — APPROVED** [review]

File: src/constants/libraryName.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+/* #if _REACTOR
+export default "https://ns.adobe.com/experience/alloy+launch";
```
Comment (jfkhoury) [inline src/constants/libraryName.js:14]: Do we need an XDM PR for this new name?

**Review by jfkhoury — APPROVED** [review]
Awesome!

File: src/constants/libraryName.js (hunk 1 above)
Comment (jonsnyder) [reply src/constants/libraryName.js:14]: Yes, https://github.com/adobe/xdm/issues/964.

**Review by ninaceban — APPROVED** [review]
👍

**Pull request merged**